tests/ConstantTest.php

```diff
@@ -998,6 +998,60 @@ class X {
                     '$c' => 'int',
                 ]
             ],
+            'bitwiseAndClassConstant' => [
+                'code' => '<?php
+                    class X {
+                        public const A = 1;
+                        public const B = 2;
+                        public const C = self::A & self::B;
+                    }
+
+                    $c = X::C;',
+                'assertions' => [
+                    '$c' => 'int',
```

Comment:
Just one more minor nitpick, could you try using eg `'$c===' => '0'` for these assertions so we make sure it fully resolves the actual value? If it doesn't work that's fine and we can go ahead and merge it, but if it does work it seems better to make sure it continues working that way in the future.

Reply:
Actually looking at this again, I think this will require additional logic to the sections you added to check for literals and evaluate them, which we can leave for a future improvement.

Reply:
Well, now that it's released it turns out this [does work](https://psalm.dev/r/b55d6ea10b). The other tests that use SimpleTypeInferer are probably enough to cover it though.